internal/ingress/controller/parser/parser.go

```diff
@@ -535,13 +545,35 @@ func (p *Parser) fillOverrides(state KongState) error {
 }
 
 func overrideService(service *Service,
+	kongIngress *configurationv1.KongIngress,
+	annotations map[string]string) {
+	if service == nil {
+		return
+	}
+	if kongIngress != nil && kongIngress.Proxy != nil {
+		overrideServiceByKongIngress(service, kongIngress)
+	}
+
+	routeAnns := annotations["configuration.konghq.com/protocol"]
```

Comment:
`routeAnns` is not a good name for this variable. Maybe as simple as `protocol`?
Also, you want to define the annotation key  as a constant in the following file:
https://github.com/Kong/kubernetes-ingress-controller/blob/effb126a68abfb6485325032c14ea1be8f90c965/internal/ingress/annotations/annotations.go